### PR TITLE
new method definition for ss2float

### PR DESCRIPTION
**Pull request by milktrader**

### CODE REVIEW

File: src/utilities.jl [hunk 1]
```diff
@@ -18,4 +18,23 @@ function ss2float(ss::Array{Any})
     end
     
     return b
-end
\ No newline at end of file
+end
+
+# for v0.4
+
+function ss2float(ss::Array{Array{_<:SubString{T<:String},1},1})
```
Comment (tonyhffong) [inline src/utilities.jl]: You mean `function ss2float{T<:AbstractString}( ss::Array{Array{T,1},1} )` ?

**Comment by tonyhffong** [comment]
What is the status on this PR?

**Comment by milktrader** [comment]
Hmm, I seem to have lost track of this somehow. Do we need it? Also, I'm not sure what test it's failing.

**Comment by tonyhffong** [comment]
in julia 0.4, I'm seeing this error:
```
julia> quandl( "CURRFX/GBPUSD" )
ERROR: MethodError: `ss2float` has no method matching ss2float(::Array{Array{_<:AbstractString,1},1})
 in timearray at /Users/tonyfong/.julia/v0.4/Quandl/src/timearray.jl:25
 in quandlget at /Users/tonyfong/.julia/v0.4/Quandl/src/api.jl:27
```
I wonder if this PR is meant to fix that.

**Comment by milktrader** [comment]
Ok, right. This is probably a good case to use the `Compat` package, which I haven't used yet. I recall now not pursuing this because it will break on the current stable release, `v0.3`.

**Comment by milktrader** [comment]
Another idea, not sure if this will work, is to put the new `ss2float.jl` in a different `utilities.jl` file, say `utilities04.jl` and control flow the `include` in the module something like this:

````julia
VERSION < v"0.4-" ? include("utilities03.jl") : include("utilities04.jl")
````
This is admittedly a bit hackish.

File: src/utilities.jl (hunk 1 above)
Comment (milktrader) [inline src/utilities.jl]: Trying this now, I'm sure this is what's needed

**Comment by milktrader** [comment]
I'm putting down my troubleshooting history since it's such a pain to get a closer look at what's going on 

```
query_args = {"sort_order" => order, "rows" => 100, "collapse" => "daily", "transformation" => "none", "trim_start" => "", "trim_end" => ""}
 auth_token = ""
id ="GOOG/NASDAQ_QQQ"
response = get("http://www.quandl.com/api/v1/datasets/$id.csv", query = query_args)
data = split(response.data, "\n")
head = data[1]
body = data[2:end]
body[end] == "" ? pop!(body) : nothing # remove tr
body      = [split(line, ",") for line in body] # split on
vals = [line[2:end] for line in body] #

# this variable calls ss2float 
function ss2float(ss::Array{SubString{ASCIIString},1} )
println("yay")
end

ss2float(vals)

function ss2float(ss::Array{Any,1} )
println("yay")
end

ss2float(vals)

function ss2float(ss::Array{Any,1} )
b = zeros(length(ss),length(ss[1]))
for r in 1:size(b,1)
for c in 1:size(b,2)
if ~isempty(ss[r][c])
b[r,c] = float(ss[r][c])
else
b[r,c] = NaN
end
````

**Comment by milktrader** [comment]
I see I've made a mistake in my compatibility hack in that I called the include files too far down the list. The hack works for now so I'll push the change, but let's not call it a *solution*

**Comment by milktrader** [comment]
For some reason I don't understand, the REPL experiment reveals a working `ss2float` method, but it fails when calling the package. 

````julia
julia> quandl("GOOG/NASDAQ_QQQ")
ERROR: `ss2float` has no method matching ss2float(::Array{Array{_<:AbstractString,1},1})
 in timearray at /Users/Administrator/.julia/v0.4/Quandl/src/timearray.jl:25
 in quandlget at /Users/Administrator/.julia/v0.4/Quandl/src/api.jl:27
````

**Comment by milktrader** [comment]
So two hacked solutions have this working on `v0.4` now. The nightlies are failing on travis, but the releases are green. 

@tonyhffong what do you think about merging this? I'm a little wary because it's such a hack to get drop type signature information to get something to work. I'm ok with the call to different utilities files though, but there is probably a better way to do this.

**Comment by milktrader** [comment]
If we merge this, I can open two issues:

* put type signature back in `ss2float` method
* find more elegant way to call different `ss2float` methods based on Julia version

**Comment by tonyhffong** [comment]
I don't understand why we don't directly do the parsing in `timearray.jl` line 14. We are holding all the intermediate substrings of the response in memory, which seem inefficient. Is ss2float being used anywhere else?

**Comment by milktrader** [comment]
Well this is failing on `v0.4` because of some issue with reading the JLD file. I can trouble shoot that later in a separate issue. Let me clean up the commented out references to ss2float and I'll merge this.

**Comment by tonyhffong** [comment]
Thank you.

**Pull request merged**